source.go

```diff
@@ -209,7 +175,7 @@ func (s *Source) Ack(ctx context.Context, position sdk.Position) error {
 		return fmt.Errorf("failed to ack message: %w", err)
 	}
 
-	s.storedMessages.remove(pos.MessageID)
+	s.storedMessages.Pop(pos.MessageID)
```

Comment:
Since we don't need the result of `Pop` we could use `Remove` instead.
```suggestion
	s.storedMessages.Remove(pos.MessageID)
```